client/components/webpack-build-monitor/index.jsx

```diff
@@ -5,147 +5,113 @@
+	if ( ! isConnected ) {
+		return 'Dev Server disconnected';
 	}
 
-	if ( startsWith( message, "[HMR] The following modules couldn't be hot updated" ) ) {
-		return true;
+	if ( hasError ) {
+		return 'Build error';
 	}
 
-	return state;
-};
-
-const store = createStore(
-	combineReducers( {
-		buildStatus: buildStatusReducer,
-		reloadRequired: reloadReducer,
-	} )
-);
+	if ( isBuildingCss && isBuildingJs ) {
+		return 'Rebuilding JS and CSS';
+	}
 
-const wrapConsole = fn => ( message, ...args ) => {
-	Promise.resolve().then( () => store.dispatch( { type: CONSOLE_MESSAGE, message } ) );
-	fn.call( window, message, ...args );
-};
+	if ( isBuildingCss ) {
+		return 'Building CSS';
+	}
 
-console.error = wrapConsole( console.error );
-console.warn = wrapConsole( console.warn );
-console.log = wrapConsole( console.log );
+	if ( isBuildingJs ) {
+		return 'Rebuilding JavaScript';
+	}
 
-const STATUS_TEXTS = {
-	[ DISCONNECTED ]: 'Dev Server disconnected',
-	[ ERROR ]: 'Build error',
-	[ BUILDING_JS ]: 'Rebuilding Javascript',
-	[ BUILDING_CSS ]: 'Rebuilding CSS',
-	[ BUILDING_BOTH ]: 'Rebuilding JS and CSS',
+	return '';
 };
 
-const RELOAD_REQUIRED_ELEMENT = (
-	// eslint-disable-next-line wpcalypso/jsx-classname-namespace
-	<div className="webpack-build-monitor is-warning">Need to refresh</div>
-);
+let alreadyExists = false;
 
 class WebpackBuildMonitor extends React.PureComponent {
-	state = {
-		buildStatus: IDLE,
-		reloadRequired: false,
-	};
+	state = buildState;
 
 	componentDidMount() {
-		this.unsubscribe = store.subscribe(
-			( () => {
-				let prevState = undefined;
-				return () => {
-					const nextState = store.getState();
-					if ( nextState !== prevState ) {
-						this.setState( nextState );
-						prevState = nextState;
-					}
-				};
-			} )()
-		);
-	}
+		if ( alreadyExists ) {
+			console.error(
+				'There should only be a single build monitor loaded. ' +
+					"Please make sure we're not trying to load more than one at a time."
+			);
+		}
+		alreadyExists = true;
 
-	componentWillUnmount() {
-		this.unsubscribe();
+		interceptConsole( console, state => this.setState( state ) );
 	}
 
 	render() {
-		const { buildStatus, reloadRequired } = this.state;
+		const { hasError, isBuildingCss, isBuildingJs, isConnected } = this.state;
 
-		if ( buildStatus === IDLE ) {
-			return reloadRequired && RELOAD_REQUIRED_ELEMENT;
+		// build is idle
+		if ( ! isBuildingCss && ! isBuildingJs ) {
+			return null;
 		}
 
-		const text = STATUS_TEXTS[ buildStatus ] || '';
-
-		const classnames = classNames( 'webpack-build-monitor', {
-			'is-error': status === ERROR || status === DISCONNECTED,
-			'is-warning': reloadRequired,
-		} );
-
 		return (
-			<div className={ classnames }>
-				{ includes( [ BUILDING_BOTH, BUILDING_CSS, BUILDING_JS ], buildStatus ) &&
+			<div
+				className={ classNames( 'webpack-build-monitor', {
+					'is-error': hasError || ! isConnected,
```

Comment:
similar comment here if you decide to add back in the NEEDS_RELOAD state.

Reply:
done as of 6aef3fd4f